numpy/core/src/multiarray/methods.c

```diff
@@ -1008,6 +1008,44 @@ array_getarray(PyArrayObject *self, PyObject *args)
 
 
 static PyObject *
+array_ufunc(PyArrayObject *self, PyObject *args, PyObject *kwds)
+{
+    PyObject *ufunc, *method_name, *normal_args, *ufunc_method, *result;
+
+    if (PyTuple_Size(args) < 2) {
+        PyErr_SetString(PyExc_TypeError,
+                        "__array_ufunc__ requires at least 2 arguments");
+        return NULL;
+    }
+    ufunc = PyTuple_GET_ITEM(args, 0);  /* borrowed reference */
+    if (ufunc == NULL) {
+        return NULL;
+    }
+
+    method_name = PyTuple_GET_ITEM(args, 1);
+    if (method_name == NULL) {
+        return NULL;
+    }
+
+    normal_args = PyTuple_GetSlice(args, 2, PyTuple_GET_SIZE(args));
+    if (normal_args == NULL) {
+        return NULL;
+    }
+
+    ufunc_method = PyObject_GetAttr(ufunc, method_name);
+    if (ufunc_method == NULL) {
+        Py_DECREF(normal_args);
+        return NULL;
+    }
+
+    result = PyObject_Call(ufunc_method, normal_args, kwds);
```

Comment:
You're not missing anything: the subclass has to clean the arguments up before calling `super`. It really just replaces `getattr(ufunc, method)`. Indeed, the main advantage over it is for a class like
```
class MaskedQuantity(MaskedArray, Quantity):
    def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
        ...
        result = super().__array_ufunc__(ufunc, method, *handled_inputs, **handled_kwargs)
        ...
        return handled_result
```
For this class, super passes on to `MaskedArray`, and `MaskedArray`'s super will not go to `ndarray` as it normally would, but to `Quantity`.